### PR TITLE
changes for macOS support

### PR DESCRIPTION
**Pull request by corbinstreehouse**
These are the minimal changes required to support macOS with BLE. I wrote the backend scanner from scratched, and commonized a bit of the code, so the diff looks large, but really isn't that big of a change. We could leave tilt_monitor_aio.py as it was, but it is always nice to share code. This is based off the branch of my other fixes, so it includes them too.

I haven't tested these changes on the pi yet; I think everything should be fine, but please do give it a run. I'll try to test it there soon too. 

Let me know what you think! My Mac setup app is now ready.. https://github.com/corbinstreehouse/FermentrackManager  -- it is pulling from my cloned repo until these changes are available.

### CODE REVIEW

File: utils/processmgr.py [hunk 1]
```diff
@@ -33,8 +33,16 @@
...
+# remove the utils folder from the end of the path
+fermentrack_install_location = os.path.split(process_mgr_location)[0]
+
+fermentrack_log_path = os.path.join(fermentrack_install_location, "log")
+
+brewpi_script_path = os.path.join(fermentrack_install_location, "brewpi-script/brewpi.py")
+
 ########## BrewPi Script Configuration
-BREWPI_SCRIPT_CMD_TEMPLATE = "python -u " + os.path.expanduser("~/fermentrack/brewpi-script/brewpi.py") + ' --dbcfg "%s"'
+BREWPI_SCRIPT_CMD_TEMPLATE = "python -u " + brewpi_script_path + ' --dbcfg "%s"'
```
Comment (thorrak) [inline utils/processmgr.py]: This is a MUCH better way of doing this.

File: utils/processmgr.py [hunk 2]
```diff
@@ -93,7 +106,7 @@ def run():
         device_type="BrewPi",
         command_tmpl=BREWPI_SCRIPT_CMD_TEMPLATE,
         circus_endpoint=DEFAULT_ENDPOINT_DEALER,
-        logfilepath=os.path.expanduser("~/fermentrack/log"),
+        logfilepath=fermentrack_log_path,
```
Comment (thorrak) [inline utils/processmgr.py]: Sooooooo much better. Thank you!

**Comment by thorrak** [comment]
I'm going to merge these into a new branch for the moment so that I can test that everything still works with the changes I made earlier today. Assuming everything works on a Pi, I'll try to get these into dev either tomorrow or Sunday

**Comment by thorrak** [comment]
Everything looks/works great - I'll get the changes pushed to dev momentarily. We should get something added to the docs for your MacOS helper app!

**Comment by corbinstreehouse** [comment]
Awesome, thanks @thorrak ! I did a "release build" of the macOS helper app (FermentrackManager) - I want to work on a few last issues and then I'll post it to the home-brew forums to get some other people to try testing it out.

**Pull request merged**